superset-frontend/src/explore/components/ExploreViewContainer/index.jsx

```diff
@@ -337,7 +375,7 @@ function ExploreViewContainer(props) {
   }
 
   function toggleModal() {
-    setShowingModal(!showingModal);
+    props.actions.setSaveModalVisible(!props.isSaveModalVisible);
```

Comment:
I think this PR might be re-triggering the annoying bug that forced us to [revert these two PRs](https://github.com/apache/superset/pull/20796):

https://user-images.githubusercontent.com/13007381/188811425-a306f8b2-4ec9-4aa1-94f7-504759ea1a6f.mov

Reply:
I don't think we were able to solve it: IIRC it seemed to happen whenever we tried to send additional Redux actions at the same time as when Explore was rendering.  [Here's the PR](https://github.com/apache/superset/pull/20704) that would have fixed it if we hadn't reverted the whole thing.  Not sure if your case is similar?  I think that also might be why Cypress is failing on yours.

Reply:
I also noticed that if I Save As and it doesn't refresh, it'll then show the modal every time even if I save it again.